StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1311CodeFixProvider.cs

```diff
@@ -43,8 +47,58 @@ public override ImmutableArray<string> GetFixableDiagnosticIds()
                     continue;
 
                 var newName = char.ToUpper(token.ValueText[0]) + token.ValueText.Substring(1);
-                context.RegisterFix(CodeAction.Create($"Rename field to '{newName}'", cancellationToken => RenameHelper.RenameSymbolAsync(document, root, token, newName, cancellationToken)), diagnostic);
+
+                var typeSyntax = this.GetParentTypeDeclaration(token);
+
+                if (typeSyntax != null)
+                {
+                    SemanticModel semanticModel = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                    var typeDeclarationSymbol = semanticModel.GetDeclaredSymbol(typeSyntax) as INamedTypeSymbol;
+                    if (typeDeclarationSymbol == null)
+                    {
+                        return;
+                    }
+
+                    while (!this.IsValidNewMemberName(typeDeclarationSymbol, newName) || newName == typeDeclarationSymbol.Name)
```

Comment:
I agree with putting the name check into the IsValidNewMemberName method. But without the while the code fix may suggest a code fix that will not compile if there are still name conflicts after the fix.